frontend/catalyst/jax_primitives.py

```diff
@@ -20,9 +20,9 @@
 
 import jax
 import numpy as np
-from jax._src import util
+from jax._src import api_util, core, source_info_util, util
```

Comment:
So from what I understand in the JAX repo everything is defined in `jax._src`, and a lot (but not all!) names are re-exported to the top-level as "public". 
For example, `jax.core.AbstractValue` in https://github.com/google/jax/blob/jaxlib-v0.4.10/jax/core.py.

Ideally, we would only access functions and classes from this public interface, but at the moment it looks like we can't get around importing some "hidden" ones.
But I think we should keep the number of those as small as possible and be explicit when we do so, whereas the changes in this file blanket import `core` from `jax._src` for example even though most of the classes we use are defined the public `jax.core`.

Reply:
If this is too much work you can also skip this comment and we can look at our usage of the JAX API in the future.

Reply:
Let's just leave that for another ticket, but I see exactly what you mean. Thanks for pointing it out!